File: 348 Sum of a square and a cube.pl
#!/usr/bin/perl

# Daniel "Trizen" Șuteu
# Date: 20 February 2017
# License: GPLv3
# https://github.com/trizen

# https://projecteuler.net/problem=348

# Runtime: 37.866s

use 5.010;
use strict;
use integer;

my %count;
my $count = 0;
my $sum   = 0;

OUTER: for (my $i = 1 ; ; ++$i) {
    for my $j (1 .. $i >> 2) {

        my $s = $i**2;
        my $c = $j**3;
        my $p = $s + $c;

        if ($p eq reverse($p)) {
            push @{$count{$p}}, "$i^2 + $j^3";
            if (@{$count{$p}} == 4) {
                $sum += $p;
                say "$p = [", join(", ", @{$count{$p}}), "]";
                last OUTER if ++$count == 5;
            }
        }
    }
}

say "Sum: $sum";

File: Julia/348 Sum of a square and a cube.jl
#!/usr/bin/julia

# Daniel "Trizen" Șuteu
# Date: 20 February 2017
# License: GPLv3
# https://github.com/trizen

# https://projecteuler.net/problem=348

# Runtime: 15.005s

function p348(n)

    count = Dict{Int64, Int64}()
    total = 0

    for i in 1:2^32
        for j in 1:(i>>2)
            s = i^2
            c = j^3
            p = s+c

            if (string(p) == reverse(string(p)))
                if haskey(count, p)
                    count[p] += 1
                else
                    count[p] = 1
                end

                if (count[p] == 4)
                    total += p
                    n -= 1
                end
            end
        end
        n == 0 && break
    end

    return total
end

println("Sum: ", p348(5))
